integration_tests/integration/searchForABooking.cy.ts

```diff
@@ -0,0 +1,140 @@
+import { addDays, format, sub } from 'date-fns'
+import TestData from '../../server/routes/testutils/testData'
+import HomePage from '../pages/home'
+import Page from '../pages/page'
+import SearchForBookingByReferencePage from '../pages/searchForBookingByReference'
+import SearchForBookingByPrisonerPage from '../pages/searchForBookingByPrisoner'
+import VisitDetailsPage from '../pages/visitDetails'
+
+context('Search for a booking by reference', () => {
+  const shortDateFormat = 'yyyy-MM-dd'
+  const longDateFormat = 'd MMMM yyyy'
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+    cy.task('stubSupportedPrisonIds')
+    cy.task('stubPrisons')
+    cy.signIn()
+  })
+
+  it('Should search via booking reference, than navigate to the summary page', () => {
+    const homePage = Page.verifyOnPage(HomePage)
+    const today = new Date()
+    const prisoner = TestData.prisoner()
+    const { prisonerNumber: offenderNo } = prisoner
+    const prisonerDisplayName = 'Smith, John'
+    const visit = TestData.visit()
+
+    const childDob = format(sub(today, { years: 5 }), shortDateFormat)
+    const contacts = [
+      TestData.contact({ personId: 1234 }),
+      TestData.contact({
+        personId: 4322,
+        firstName: 'Bob',
+        dateOfBirth: childDob,
+        relationshipCode: 'SON',
+        relationshipDescription: 'Son',
+      }),
+    ]
+
+    homePage.changeAVisitTile().click()
+
+    const searchForBookingByReferencePage = Page.verifyOnPage(SearchForBookingByReferencePage)
+
+    searchForBookingByReferencePage.enterVisitReference('ab-cd-ef-gh')
+
+    cy.task('stubPrisonerById', prisoner)
+    cy.task('stubVisit', visit)
+
+    searchForBookingByReferencePage.continueButton().click()
+
+    searchForBookingByReferencePage.visitReference().contains('ab-cd-ef-gh')
```

Comment:
I think this is a new `Page` we're on now after the `.click()` - e.g. something like `searchForBookingByReferenceResults`. E.g. the way `pages/searchForAPrisoner.ts` and `pages/searchForAPrisonerResults.ts` are different.